micro-rdk-nmea/src/messages/pgns.rs

```diff
@@ -145,41 +145,41 @@ macro_rules! define_pgns {
                 Ok(match pgn {
                     $($pgn => {
                         let cursor = DataCursor::new(bytes);
-                        Self::$enum($pgndef::from_cursor(cursor)?)
+                        Self::$pgndef($pgndef::from_cursor(cursor)?)
                     }),*,
-                    x => Self::Unsupported(UnparsedMessageData::from_bytes(bytes, x)?)
+                    x => Self::Unsupported(UnparsedNmeaMessageBody::from_bytes(bytes, x)?)
                 })
             }
 
             pub fn to_readings(self) -> Result<GenericReadingsResult, crate::parse_helpers::errors::NmeaParseError> {
                 match self {
-                    $(Self::$enum(msg) => msg.to_readings()),*,
+                    $(Self::$pgndef(msg) => msg.to_readings()),*,
                     Self::Unsupported(msg) => msg.to_readings()
                 }
             }
         }
     };
 }
 
-pub const MESSAGE_HEADER_OFFSET: usize = 32;
+pub const MESSAGE_DATA_OFFSET: usize = 32;
 
 define_pgns!(
-    (WaterDepth, Pgn128267, 128267),
-    (TemperatureExtendedRange, Pgn130316, 130316),
-    (GnssSatsInView, Pgn129540, 129540)
+    (WaterDepth, 128267),
```

Comment:
Last question, I promise: Do you even need the PGN code as the second argument anymore?

Reply:
yes I cannot use information only available at runtime to generate the content of the match statement for the `from_bytes` function